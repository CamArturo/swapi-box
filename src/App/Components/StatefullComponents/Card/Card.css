.card {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 20px;
  padding: 0;
  -webkit-box-shadow: 4px 4px 5px 0 rgba(10, 10, 10, 0.65);
  -moz-box-shadow: 4px 4px 5px 0 rgba(10, 10, 10, 0.65);
  box-shadow: 4px 4px 5px 0 rgba(10, 10, 10, 0.65);
  border-radius: 8px;
  background-color: #2a2a57;
  color: #fff;
  height: 350px;
  width: 250px; }

.card-header {
  text-align: center;
  font-size: 1.2rem;
  height: 100px;
  margin: 0;
  padding: 0 .5rem 0 .5rem;
  letter-spacing: 1px;
  display: flex;
  align-items: center; }
  .card-header h2 {
    margin: 0; }

.card-body {
  width: 100%;
  background-color: #2a2a57;
  height: 100vh;
  padding: 1rem;
  border-bottom-left-radius: 5px;
  border-bottom-right-radius: 5px; }

/* Large Devices, Wide Screens */
/* Medium Devices, Desktops */
/* Small Devices, Tablets */
/* Extra Small Devices, Phones */
/* Custom, iPhone Retina */

/*# sourceMappingURL=Card.css.map */
