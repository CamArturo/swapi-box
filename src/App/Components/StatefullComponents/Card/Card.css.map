{
"version": 3,
"mappings": "AAAA,KAAM;EACJ,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,MAAM;EACtB,eAAe,EAAE,MAAM;EACvB,WAAW,EAAE,MAAM;EACnB,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,CAAC;EACV,kBAAkB,EAAE,oCAAoC;EACxD,eAAe,EAAE,oCAAoC;EACrD,UAAU,EAAE,oCAAoC;EAChD,aAAa,EAAE,GAAG;EAClB,gBAAgB,EAAE,OAAO;EACzB,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,KAAK;EACb,KAAK,EAAE,KAAK;;AAGd,YAAa;EACX,UAAU,EAAE,MAAM;EAClB,SAAS,EAAE,MAAM;EACjB,MAAM,EAAE,KAAK;EACb,MAAM,EAAE,CAAC;EACT,OAAO,EAAE,eAAe;EACxB,cAAc,EAAE,GAAG;EACnB,OAAO,EAAE,IAAI;EACb,WAAW,EAAE,MAAM;EACnB,eAAG;IACD,MAAM,EAAE,CAAC;;AAIb,UAAW;EACT,KAAK,EAAE,IAAI;EACX,gBAAgB,EAAE,OAAO;EACzB,MAAM,EAAE,KAAK;EACb,OAAO,EAAE,IAAI;EACb,yBAAyB,EAAE,GAAG;EAC9B,0BAA0B,EAAE,GAAG;;AAGjC,iCAAiC;AAKjC,8BAA8B;AAK9B,4BAA4B;AAM5B,iCAAiC;AAKjC,2BAA2B",
"sources": ["Card.scss"],
"names": [],
"file": "Card.css"
}