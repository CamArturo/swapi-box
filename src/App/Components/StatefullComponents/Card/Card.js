import React, {Component} from 'react';
import PropTypes from 'prop-types';

class Card extends Component {
  constructor(props) {
    super(props);

  }

  render() {
    return (
      <article className="card">
        <p>Luke Skywalker</p>
        <p>HomeWorld: Tatooine</p>
        <p>Species: Human</p>
        <p>Population: 200000</p>
        <button className="favorite">favorite</button>
      </article>
    );
  }
}

Card.propTypes = {};

export default Card;
