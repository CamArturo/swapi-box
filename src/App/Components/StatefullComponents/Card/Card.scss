.card {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 20px;
  padding: 0;
  -webkit-box-shadow: 4px 4px 5px 0 rgba(10, 10, 10, 0.65);
  -moz-box-shadow: 4px 4px 5px 0 rgba(10, 10, 10, 0.65);
  box-shadow: 4px 4px 5px 0 rgba(10, 10, 10, 0.65);
  border-radius: 8px;
  background-color: #2a2a57;
  color: #fff;
  height: 350px;
  width: 250px;
}

.card-header {
  text-align: center;
  font-size: 1.2rem;
  height: 80px;
  margin: 0;
  padding: 0 .5rem 0 .5rem;
  letter-spacing: 1px;
  display: flex;
  align-items: center;
  h2 {
    margin: 0;
  }
}

.card-body {
  width: 100%;
  background-color: #fff;
  text-align: center;
  color: #2a2a57;
  height: 270px;
  padding: 1rem;
  border-bottom-left-radius: 5px;
  border-bottom-right-radius: 5px;
  p {
    min-height: 36px;
  }
}

.favorite-container {
  display: flex;
  align-content: baseline;
  justify-content: center;
  .favorite {
    margin: 0 1rem;
    background-color: #2a2a57;
    color: #fff;
    border-radius: 5px;
    width: 90px;
    padding: 10px;
  }
}

/* Large Devices, Wide Screens */
@media only screen and (max-width: 1200px) {

}

/* Medium Devices, Desktops */
@media only screen and (max-width: 992px) {

}

/* Small Devices, Tablets */
@media only screen and (max-width: 768px) {

}

/* Extra Small Devices, Phones */
@media only screen and (max-width: 480px) {

}

/* Custom, iPhone Retina */
@media only screen and (max-width: 320px) {

}