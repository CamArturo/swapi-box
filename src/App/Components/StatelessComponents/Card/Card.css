.card {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 20px;
  padding: 4px;
  -webkit-box-shadow: 4px 4px 5px 0 rgba(10, 10, 10, 0.65);
  -moz-box-shadow: 4px 4px 5px 0 rgba(10, 10, 10, 0.65);
  box-shadow: 4px 4px 5px 0 rgba(10, 10, 10, 0.65);
  border-radius: 8px;
  background-color: #2a2a57;
  color: #fff;
  height: 390px;
  width: 250px; }
  .card.selected {
    border: 4px solid #FFD100; }
  .card .card-header {
    text-align: center;
    font-size: 1.2rem;
    height: 120px;
    margin: 0;
    padding: 0 .5rem 0 .5rem;
    letter-spacing: 1px;
    display: flex;
    align-items: center; }
    .card .card-header h2 {
      margin: 0;
      text-transform: capitalize; }
  .card .card-body {
    width: 100%;
    background-color: #fff;
    text-align: center;
    color: #2a2a57;
    height: 310px;
    padding: 1rem;
    position: relative;
    border-bottom-left-radius: 5px;
    border-bottom-right-radius: 5px; }
    .card .card-body .last {
      height: 56px;
      overflow: hidden; }
    .card .card-body p {
      text-transform: capitalize; }

.card.selected {
  padding: 0; }

.favorite-container {
  position: absolute;
  top: calc(100% - 68px);
  width: calc(100% - 2rem); }
  .favorite-container .favorite {
    margin: 0 1rem;
    background-color: #2a2a57;
    color: #fff;
    width: 90px;
    padding: 10px; }
    .favorite-container .favorite:hover {
      background-color: #414141;
      border: 1px solid #fff; }

/* Large Devices, Wide Screens */
/* Medium Devices, Desktops */
/* Small Devices, Tablets */
/* Extra Small Devices, Phones */
/* Custom, iPhone Retina */
