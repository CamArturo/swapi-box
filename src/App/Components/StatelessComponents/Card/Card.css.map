{
"version": 3,
"mappings": "AAAA,KAAM;EACJ,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,MAAM;EACtB,eAAe,EAAE,MAAM;EACvB,WAAW,EAAE,MAAM;EACnB,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,GAAG;EACZ,kBAAkB,EAAE,oCAAoC;EACxD,eAAe,EAAE,oCAAoC;EACrD,UAAU,EAAE,oCAAoC;EAChD,aAAa,EAAE,GAAG;EAClB,gBAAgB,EAAE,OAAO;EACzB,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,KAAK;EACb,KAAK,EAAE,KAAK;EACZ,cAAW;IACT,MAAM,EAAE,iBAAiB;EAE3B,kBAAa;IACX,UAAU,EAAE,MAAM;IAClB,SAAS,EAAE,MAAM;IACjB,MAAM,EAAE,KAAK;IACb,MAAM,EAAE,CAAC;IACT,OAAO,EAAE,eAAe;IACxB,cAAc,EAAE,GAAG;IACnB,OAAO,EAAE,IAAI;IACb,WAAW,EAAE,MAAM;IACnB,qBAAG;MACD,MAAM,EAAE,CAAC;MACT,cAAc,EAAE,UAAU;EAG9B,gBAAW;IACT,KAAK,EAAE,IAAI;IACX,gBAAgB,EAAE,IAAI;IACtB,UAAU,EAAE,MAAM;IAClB,KAAK,EAAE,OAAO;IACd,MAAM,EAAE,KAAK;IACb,OAAO,EAAE,IAAI;IACb,QAAQ,EAAE,QAAQ;IAClB,yBAAyB,EAAE,GAAG;IAC9B,0BAA0B,EAAE,GAAG;IAC/B,sBAAM;MACJ,MAAM,EAAE,IAAI;MACZ,QAAQ,EAAE,MAAM;IAElB,kBAAE;MACA,cAAc,EAAE,UAAU;;AAKhC,cAAe;EACb,OAAO,EAAE,CAAC;;AAGZ,mBAAoB;EAClB,QAAQ,EAAE,QAAQ;EAClB,GAAG,EAAE,iBAAiB;EACtB,KAAK,EAAE,iBAAiB;EACxB,6BAAU;IACR,MAAM,EAAE,MAAM;IACd,gBAAgB,EAAE,OAAO;IACzB,KAAK,EAAE,IAAI;IAEX,KAAK,EAAE,IAAI;IACX,OAAO,EAAE,IAAI;IACb,mCAAQ;MACN,gBAAgB,EAAE,OAAO;MACzB,MAAM,EAAE,cAAc;;AAK5B,iCAAiC;AAKjC,8BAA8B;AAK9B,4BAA4B;AAK5B,iCAAiC;AAKjC,2BAA2B",
"sources": ["Card.scss"],
"names": [],
"file": "Card.css"
}