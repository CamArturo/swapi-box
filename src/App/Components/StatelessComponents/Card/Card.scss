.card {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 20px;
  padding: 4px;
  -webkit-box-shadow: 4px 4px 5px 0 rgba(10, 10, 10, 0.65);
  -moz-box-shadow: 4px 4px 5px 0 rgba(10, 10, 10, 0.65);
  box-shadow: 4px 4px 5px 0 rgba(10, 10, 10, 0.65);
  border-radius: 8px;
  background-color: #2a2a57;
  color: #fff;
  height: 390px;
  width: 250px;
  &.selected {
    border: 4px solid #FFD100;
  }
  .card-header {
    text-align: center;
    font-size: 1.2rem;
    height: 120px;
    margin: 0;
    padding: 0 .5rem 0 .5rem;
    letter-spacing: 1px;
    display: flex;
    align-items: center;
    h2 {
      margin: 0;
      text-transform: capitalize;
    }
  }
  .card-body {
    width: 100%;
    background-color: #fff;
    text-align: center;
    color: #2a2a57;
    height: 310px;
    padding: 1rem;
    position: relative;
    border-bottom-left-radius: 5px;
    border-bottom-right-radius: 5px;
    .last {
      height: 56px;
      overflow: hidden;
    }
    p {
      text-transform: capitalize;
    }
  }
}

.card.selected {
  padding: 0;
}

.favorite-container {
  position: absolute;
  top: calc(100% - 68px);
  width: calc(100% - 2rem);
  .favorite {
    margin: 0 1rem;
    background-color: #2a2a57;
    color: #fff;
    //border-radius: 5px;
    width: 90px;
    padding: 10px;
    &:hover {
      background-color: #414141;
      border: 1px solid #fff;
    }
  }
}

/* Large Devices, Wide Screens */
@media only screen and (max-width: 1200px) {

}

/* Medium Devices, Desktops */
@media only screen and (max-width: 992px) {

}

/* Small Devices, Tablets */
@media only screen and (max-width: 768px) {

}

/* Extra Small Devices, Phones */
@media only screen and (max-width: 480px) {

}

/* Custom, iPhone Retina */
@media only screen and (max-width: 320px) {

}