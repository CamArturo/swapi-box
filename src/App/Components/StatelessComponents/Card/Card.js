import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';

const Card = ({ cardInfo, category, updateFavorites }) => {
  const keys = Object.keys(cardInfo);
  const second = keys[1];
  const third = keys[2];
  const fourth = keys[3];
  const fifth = keys[4];
  return (
    <article className="card">
      <section className="card-header">
        <h2>{cardInfo.name}</h2>
      </section>
      <section className="card-body">
        <p>{second} : {cardInfo[second]}</p>
        <p>{third} : {cardInfo[third]}</p>
        <p>{fourth} : {cardInfo[fourth]}</p>
        <p>{cardInfo[fifth]}</p>
        <div className="favorite-container">
          <button className="favorite" onClick={() => {
            updateFavorites(cardInfo, category);
          }}>favorite</button>
        </div>
      </section>
    </article>
  );
};

Card.propTypes = {
  updateFavorites: PropTypes.func.isRequired,
  category: PropTypes.string.isRequired,
  cardInfo: PropTypes.object.isRequired
};

export default Card;
