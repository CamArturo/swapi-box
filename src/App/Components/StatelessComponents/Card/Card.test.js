import React from 'react';
import {shallow} from 'enzyme';
import Card from "./Card";

describe('Card test', () => {

  const updateFavorites = jest.fn();
  const category = 'people';
  const card = {
    favorite: false,
    homeworld: "Tatooine",
    name: "Luke Skywalker",
    population: "200000",
    species: "Human",
    category: "people",
    selectedClass: "card"
  };
  const selectedClass = 'card selected';

  it('should match the snapshot', () => {

    const wrapper = shallow(<Card
      selectedClass={selectedClass}
      cardInfo={card}
      category={category}
      updateFavorites={updateFavorites}
    />);

    expect(wrapper).toMatchSnapshot();
  });

// test methods that are in render or lifecycle methods they are going to fire.
  // fetch is called with params
  // any function that act upon response of the fetch
  // handle errors handle correctly

  // a did you write
  // b is it important
});
