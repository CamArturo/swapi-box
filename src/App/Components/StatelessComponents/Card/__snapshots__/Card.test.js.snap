// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card test should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardInfo={
        Object {
            "category": "people",
            "favorite": false,
            "homeworld": "Tatooine",
            "name": "Luke Skywalker",
            "population": "200000",
            "selectedClass": "card",
            "species": "Human",
          }
    }
    category="people"
    selectedClass="card selected"
    updateFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="card-header"
>
          <h2>
                    Luke Skywalker
          </h2>
</section>,
        <section
          className="card-body"
>
          <p>
                    homeworld
                     : 
                    Tatooine
          </p>
          <p>
                    name
                     : 
                    Luke Skywalker
          </p>
          <p>
                    population
                     : 
                    200000
          </p>
          <p
                    className="last"
          >
                    Human
          </p>
          <div
                    className="favorite-container"
          >
                    <button
                              className="favorite"
                              onClick={[Function]}
                    >
                              favorite
                    </button>
          </div>
</section>,
      ],
      "className": "card selected",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            Luke Skywalker
</h2>,
          "className": "card-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Luke Skywalker",
          },
          "ref": null,
          "rendered": "Luke Skywalker",
          "type": "h2",
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              homeworld
               : 
              Tatooine
</p>,
            <p>
              name
               : 
              Luke Skywalker
</p>,
            <p>
              population
               : 
              200000
</p>,
            <p
              className="last"
>
              Human
</p>,
            <div
              className="favorite-container"
>
              <button
                            className="favorite"
                            onClick={[Function]}
              >
                            favorite
              </button>
</div>,
          ],
          "className": "card-body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "homeworld",
                " : ",
                "Tatooine",
              ],
            },
            "ref": null,
            "rendered": Array [
              "homeworld",
              " : ",
              "Tatooine",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "name",
                " : ",
                "Luke Skywalker",
              ],
            },
            "ref": null,
            "rendered": Array [
              "name",
              " : ",
              "Luke Skywalker",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "population",
                " : ",
                "200000",
              ],
            },
            "ref": null,
            "rendered": Array [
              "population",
              " : ",
              "200000",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Human",
              "className": "last",
            },
            "ref": null,
            "rendered": "Human",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="favorite"
                onClick={[Function]}
>
                favorite
</button>,
              "className": "favorite-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "favorite",
                "className": "favorite",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "favorite",
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "section",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="card-header"
>
            <h2>
                        Luke Skywalker
            </h2>
</section>,
          <section
            className="card-body"
>
            <p>
                        homeworld
                         : 
                        Tatooine
            </p>
            <p>
                        name
                         : 
                        Luke Skywalker
            </p>
            <p>
                        population
                         : 
                        200000
            </p>
            <p
                        className="last"
            >
                        Human
            </p>
            <div
                        className="favorite-container"
            >
                        <button
                                    className="favorite"
                                    onClick={[Function]}
                        >
                                    favorite
                        </button>
            </div>
</section>,
        ],
        "className": "card selected",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2>
              Luke Skywalker
</h2>,
            "className": "card-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Luke Skywalker",
            },
            "ref": null,
            "rendered": "Luke Skywalker",
            "type": "h2",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                homeworld
                 : 
                Tatooine
</p>,
              <p>
                name
                 : 
                Luke Skywalker
</p>,
              <p>
                population
                 : 
                200000
</p>,
              <p
                className="last"
>
                Human
</p>,
              <div
                className="favorite-container"
>
                <button
                                className="favorite"
                                onClick={[Function]}
                >
                                favorite
                </button>
</div>,
            ],
            "className": "card-body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "homeworld",
                  " : ",
                  "Tatooine",
                ],
              },
              "ref": null,
              "rendered": Array [
                "homeworld",
                " : ",
                "Tatooine",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "name",
                  " : ",
                  "Luke Skywalker",
                ],
              },
              "ref": null,
              "rendered": Array [
                "name",
                " : ",
                "Luke Skywalker",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "population",
                  " : ",
                  "200000",
                ],
              },
              "ref": null,
              "rendered": Array [
                "population",
                " : ",
                "200000",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Human",
                "className": "last",
              },
              "ref": null,
              "rendered": "Human",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="favorite"
                  onClick={[Function]}
>
                  favorite
</button>,
                "className": "favorite-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "favorite",
                  "className": "favorite",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "favorite",
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "section",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
